test/shopify-cli/core/monorail_test.rb

```diff
@@ -77,6 +81,46 @@ def test_log_event_contains_schema_and_payload_values
         end
       end
 
+      def test_log_event_handles_errors
+        enabled_and_consented(true, true)
+        Timecop.freeze do |time|
+          this_time = (time.utc.to_f * 1000).to_i
+          stub_request(:post, Monorail::ENDPOINT_URI)
+            .with(
+              headers: {
+                'Content-Type': 'application/json; charset=utf-8',
+                'X-Monorail-Edge-Event-Created-At-Ms': this_time.to_s,
+                'X-Monorail-Edge-Event-Sent-At-Ms': this_time.to_s,
+              },
+              body: JSON.dump({
+                schema_id: ShopifyCli::Core::Monorail::INVOCATIONS_SCHEMA,
+                payload: {
+                  project_type: 'fake',
+                  args: "testcommand arg argtwo",
+                  time_start: this_time,
+                  time_end: this_time,
+                  total_time: 0,
+                  success: false,
+                  error_message: 'test error',
+                  uname: RbConfig::CONFIG["host"],
+                  cli_sha: "bb6f42193239a248f054e5019e469bc75f3adf1b",
+                  ruby_version: RUBY_VERSION,
+                  api_client_id: "apikey",
+                  partner_id: 42,
+                },
+              })
+            )
+            .to_return(status: 200)
+
+          begin
+            ShopifyCli::Core::Monorail.log('testcommand', %w(arg argtwo)) do
+              raise 'test error'
+            end
+          rescue
```

Comment:
No, I am not asserting that it is raising, that is not the concern of this test, that is tested in another test. This test is testing an error payload. This rescue is here simply because I needed to raise inside log but log also propagates the error up. I *can* change it to an assert but as I said, it really isn't the concern of this test.

Reply:
ah gotcha! thanks for the explanation. Maybe raise something specific and catch exactly that at the risk of swallowing other errors that should have made the test fail?  nitpicky though. I'll leave it up to you 👍